### PR TITLE
#3782 Cleanup realm schemas

### LINKED ISSUE #3782: Clean up app-db and data-store schemas
With the new protocol the code has been greatly improved and simplified, we should now review the current data representations that we have and clean up the app-db and data-store schemas, removing duplications and using better names.

Exemples:
- current-public-key in app-db is a duplicate of the public-key of current-account
- public-key and whisper-id in contacts are the same, we need to decide which one to keep

This issue will include upgrading realm to version 2.3.3 which includes various changes:
- Added property Realm.isClosed which indicates if a Realm instance is closed or not.
- Potential performance enhancements in cases of many writes between queries.
- [Sync] A use-after-free bug was fixed which could cause arrays of primitives to behave unexpectedly.
- [Sync] Fixed race condition in handling of session bootstrapping in client.
- Fixed a bug that could result in notifications from the global notifier being delayed or not delivered at all when multiple Realms change at once.

Also we previously didn't take advantage of realm support for arrays of primitives which was leading to the creation of unecessary schemas and parsing when reading/writing

### PR DESCRIPTION
**Pull request by yenda**
partially fulfils #3782, more changes might be done in Jan PR related to contact refactoring
 depends on https://github.com/status-im/status-react/pull/3791

status: ready

### CODE REVIEW

**Comment by janherich** [comment]
Please don't merge this, as I will open comprehensive contact refactoring branch very soon with similar changes

File: src/status_im/data_store/realm/chats.cljs [hunk 1]
```diff
@@ -82,35 +82,23 @@
...
+                                                   (realm/js-object->clj contacts))))))))
 
 (defn remove-contacts
   [chat-id identities]
-  (let [contacts (get-contacts chat-id)]
-    (delete-contacts identities contacts)))
+  (let [chat     (get-by-id-obj chat-id)
+        contacts (object/get chat "contacts")]
+    (realm/write @realm/account-realm
+                 #(aset chat "contacts"
```
Comment (jeluard) [inline src/status_im/data_store/realm/chats.cljs:99]: We should consider helper fns for those interop bits.

File: src/status_im/ui/screens/add_new/views.cljs [hunk 2]
```diff
@@ -22,12 +22,12 @@
      :on-press            #(re-frame/dispatch [:navigate-to :new-chat])}]
    [action-button/action-separator]
    ;; TODO temporary removal before everything is fixed in group chats
-   #_[action-button/action-button
-      {:label               (i18n/label :t/start-group-chat)
-       :accessibility-label :start-group-chat-button
-       :icon                :icons/contacts
-       :icon-opts           {:color colors/blue}
-       :on-press            #(re-frame/dispatch [:open-contact-toggle-list :chat-group])}]
+   [action-button/action-button
```
Comment (jeluard) [inline src/status_im/ui/screens/add_new/views.cljs:25]: Is this related?

**Review by jeluard — APPROVED** [review]

**Review by janherich — CHANGES_REQUESTED** [review]
Please wait till my contact/contact-groups refactoring PR is merged, then this can be deprecated

**Pull request closed without merging**